src/Microsoft.DotNet.Compiler.Common/Executable.cs

```diff
@@ -72,6 +73,35 @@ private void MakeCompilationOutputRunnableForFullFramework()
             CopyAssets(dependencies);
             GenerateBindingRedirects(_exporter);
         }
+        
+        private IEnumerable<string> FindCoreHostFiles()
+        {
+            if (string.IsNullOrEmpty(_context.RuntimeIdentifier))
+            {
+                yield break;
+            }
+
+            string[] allowedCoreHostFileNames = new string[] {
+                "dotnet",
+                "hostpolicy",
+                "hostfxr",
+                "libhostpolicy",
+                "libhostfxr"
+            };
+            foreach (var export in _exporter.GetAllExports())
```

Comment:
What if we pass in the exports here, so this function is less dependent on class state.